src/sqlfluff/rules/L009.py

```diff
@@ -106,12 +106,13 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
             select_if=sp.is_type("newline"),
             loop_while=sp.or_(sp.is_whitespace(), sp.is_type("dedent")),
         )
-        trailing_literal_newlines = trailing_newlines.select(
```

Comment:
Unrelated fix to a bug that surfaced during testing. Some of the automated tests don't provide `context.templated_file`, so the rule was throwing a runtime error trying to use it.